src/usecase_clustering.Rmd

```diff
@@ -0,0 +1,148 @@
+
+
+```{r, warning=FALSE, eval = FALSE}
+listLearners(obj = agri.task)
+```
+
+We will apply the $k$-means algorithm with $3$ centers for the moment
+
+```{r}
+cluster.lrn = makeLearner("cluster.kmeans", centers = 3)
+cluster.lrn
+```
+### Train the model
+
+The next step is to train our learner by feeding it with our data.
+
+```{r}
+agri.mod = train(learner = cluster.lrn, task = agri.task)
+```
+
+We can extract the model and have a look at it. 
+
+```{r}
+getLearnerModel(agri.mod)
+```
+
+### Prediction
+
+Now, we can predict the target values, our cluster labels.  
+
+```{r}
+agri.pred = predict(agri.mod, task = agri.task)
+agri.pred
+```
+
+### Performance 
+
+Since the data given to the learner is unlabeled, there is no objective evaluation of the accuracy of our model. We have to consider other criterions in unsupervised learning. 
+
+An overview over all performance measures can be found  [here](http://mlr-org.github.io/mlr-tutorial/devel/html/measures/index.html). You can also call the \texttt{listMeasures} command for our specific task.
+
+```{r}
+listMeasures(agri.task)
+```
+
+Let's have a look at the silhouette coefficient and the Davies-Boulding index.
+
+```{r message = FALSE}
+library("clValid")
+
+performance(agri.pred, measures = list(silhouette, db), task = agri.task)
+```
+
+### Tuning
+
+It's hard to say if our clustering is good since up to now we have nothing to compare to. Could we have done better by choosing another parameter for the number of centers? 
+
+Tuning will address the question of choosing the best hyperparameters for our problem.
+
+We first create a search space for the number of clusters $k$, e. g. $k \in \lbrace 2, 3, 4, 5 \rbrace$. Further we define an optimization algorithm and a [resampling strategy](http://mlr-org.github.io/mlr-tutorial/devel/html/resample/index.html).
+
+Finally, by combining all the previous pieces, we can tune the parameter $k$ by calling \texttt{tuneParams}. We will use discrete_ps with grid search and the silhouette coefficient as optimization criterion:
+
+```{r}
+discrete_ps = makeParamSet(makeDiscreteParam("centers", values = c(2, 3, 4, 5)))
+ctrl = makeTuneControlGrid()
+res = tuneParams(cluster.lrn, agri.task, measures = silhouette, resampling = cv3, 
+                 par.set = discrete_ps, control = ctrl)
+```
+
+Setting $k=2$ yields the best results for our clustering problem. 
+So let's generate a learner with the optimal hyperparameter $k=2$. 
+
+```{r}
+tuned.lrn = setHyperPars(cluster.lrn, par.vals = res$x)
+```
+
+We have to train the tuned learner again and predict the results.
+```{r}
+tuned.mod = train(tuned.lrn, agri.task)
+tuned.pred = predict(tuned.mod, task = agri.task)
+tuned.pred
+```
+
+This is our final clustering for our problem.
+
+```{r, fig.width= 5}
+plot(y ~ x, col = tuned.pred$data$response, data = agriculture)
```

Comment:
Could you please use the getter function (I think `getPredictionResponse` should work here)?